### PR TITLE
Epmlstrtai 3

### PR DESCRIPTION
**Pull request by Crash1048576**
implimented mock-service

### CODE REVIEW

File: Java/JDI/jdi-mock-rest/pom.xml [hunk 1]
```diff
@@ -0,0 +1,54 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+    xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+
+    <groupId>com.epam.jdi</groupId>
+    <artifactId>jdi-mock-rest</artifactId>
+    <version>1.0-SNAPSHOT</version>
```
Comment (EvilXa0S) [inline Java/JDI/jdi-mock-rest/pom.xml]: As far as I see most of the modules do not have version (perhaps inherit from parent) - please try without version.
Same about <goupId>.

File: Java/JDI/jdi-mock-rest/pom.xml [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+    <parent>
```
Comment (EvilXa0S) [inline Java/JDI/jdi-mock-rest/pom.xml]: Can we make parent module from JDI, and just have needed pom chages that comes from this parent?

File: Java/Tests/jdi-uitest-webtests/src/main/resources/driver/temp/LATEST_RELEASE.txt [hunk 3]
```diff
@@ -1 +0,0 @@
-2.31
```
Comment (EvilXa0S) [inline Java/Tests/jdi-uitest-webtests/src/main/resources/driver/temp/LATEST_RELEASE.txt:1]: Why changes in not related modules?

File: Java/JDI/jdi-mock-rest/src/main/java/com/epam/mock/Application.java [hunk 4]
```diff
@@ -0,0 +1,12 @@
+package com.epam.mock;
+
+import org.springframework.boot.SpringApplication;
+import org.springframework.boot.autoconfigure.SpringBootApplication;
+
+@SpringBootApplication
+public class Application {
```
Comment (EvilXa0S) [inline Java/JDI/jdi-mock-rest/src/main/java/com/epam/mock/Application.java]: Add java docs to all (ALLL) public methods/classes

File: Java/Tests/jdi-uitest-webtests/src/main/resources/driver/temp/LATEST_RELEASE.txt (hunk 3 above)
Comment (Naryadchikov) [reply Java/Tests/jdi-uitest-webtests/src/main/resources/driver/temp/LATEST_RELEASE.txt:1]: There is no such file in the project repository :)

Comment (EvilXa0S) [reply Java/Tests/jdi-uitest-webtests/src/main/resources/driver/temp/LATEST_RELEASE.txt:1]: No it is there.. It was deleted.. I assume during build

File: Java/JDI/jdi-mock-rest/pom.xml [hunk 5]
```diff
@@ -0,0 +1,73 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+    xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+
+    <groupId>com.epam.jdi</groupId>
```
Comment (EvilXa0S) [inline Java/JDI/jdi-mock-rest/pom.xml]: I assume this should be deleted also (inherited from parent as version does)

File: Java/JDI/jdi-mock-rest/pom.xml [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+    <artifactId>jdi-mock-rest</artifactId>
+
+    <properties>
+        <java.version>1.8</java.version>
+        <maven.compiler.source>${java.version}</maven.compiler.source>
```
Comment (EvilXa0S) [inline Java/JDI/jdi-mock-rest/pom.xml]: Both properties from this line and next are configured in parent pom (and have same values). And I do not see where these are used.

File: Java/JDI/jdi-mock-rest/pom.xml [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+        <maven.compiler.target>${java.version}</maven.compiler.target>
+    </properties>
+
+    <dependencyManagement>
+        <dependencies>
+            <dependency>
+                <groupId>org.springframework.data</groupId>
+                <artifactId>spring-data-releasetrain</artifactId>
```
Comment (EvilXa0S) [inline Java/JDI/jdi-mock-rest/pom.xml]: why do we need spring data?

File: Java/JDI/jdi-mock-rest/src/main/java/com/epam/mock/Application.java [hunk 8]
```diff
@@ -0,0 +1,15 @@
...
+import org.springframework.boot.SpringApplication;
+import org.springframework.boot.autoconfigure.SpringBootApplication;
+
+/**
+ * Spring Boot Application class.
+ */
+@SpringBootApplication
+public class Application {
+
+    public static void main(String[] args) {
```
Comment (EvilXa0S) [inline Java/JDI/jdi-mock-rest/src/main/java/com/epam/mock/Application.java]: please add Javadoc

File: Java/JDI/jdi-mock-rest/src/main/java/com/epam/mock/service/ConfigurationHandlerService.java [hunk 9]
```diff
@@ -0,0 +1,98 @@
...
+        configurationHandlerMap.put("updateuserfield", new ConfigurationHandler("updateuserfield", 200, ""));
+        configurationHandlerMap.put("clearuser", new ConfigurationHandler("clearuser", 404, ""));
+        configurationHandlerMap.put("clearall", new ConfigurationHandler("clearall", 200, ""));
+    }
+
+    /**
+     * Responses to controller call for all users.
+     * @return Our representation of response with JSON format.
+     */
+    public ConfigurationHandler getUsers() {
```
Comment (EvilXa0S) [inline Java/JDI/jdi-mock-rest/src/main/java/com/epam/mock/service/ConfigurationHandlerService.java]: Method name misleading. I propose to rename for example to getConfigForGetUser(). Here and in all further methods in this class (editMap() looks ok)

File: Java/JDI/jdi-mock-rest/src/main/java/com/epam/mock/service/ConfigurationHandlerService.java [hunk 10]
```diff
@@ -0,0 +1,98 @@
...
+        return configurationHandlerMap.get("getusers");
```
Comment (EvilXa0S) [inline Java/JDI/jdi-mock-rest/src/main/java/com/epam/mock/service/ConfigurationHandlerService.java]: "getusers" - extract to constant at the beginning of the class ... or even introduce Enum with applicable methods. Here and in all further methods in this class.

File: Java/Tests/jdi-uitest-webtests/src/main/resources/driver/temp/LATEST_RELEASE.txt (hunk 3 above)
Comment (EvilXa0S) [reply Java/Tests/jdi-uitest-webtests/src/main/resources/driver/temp/LATEST_RELEASE.txt:1]: You can see it on other branches - it is there (may be got deleted with the build)

File: Java/JDI/jdi-mock-rest/src/main/java/com/epam/mock/web/ConfigurationHandlerController.java [hunk 11]
```diff
@@ -0,0 +1,123 @@
...
+@RestController
+public class ConfigurationHandlerController {
+
+    private ConfigurationHandlerService configurationHandlerService;
+
+    private static final HttpHeaders httpHeaders;
+
+    static {
+        httpHeaders = new HttpHeaders();
+        httpHeaders.setContentType(MediaType.APPLICATION_JSON);
```
Comment (EvilXa0S) [inline Java/JDI/jdi-mock-rest/src/main/java/com/epam/mock/web/ConfigurationHandlerController.java:27]: Why JSON? should not this be configurable?

File: Java/Tests/jdi-uitest-webtests/src/main/resources/driver/temp/LATEST_RELEASE.txt (hunk 3 above)
Comment (Naryadchikov) [reply Java/Tests/jdi-uitest-webtests/src/main/resources/driver/temp/LATEST_RELEASE.txt:1]: I've just found that missing file and added it to our branch

File: Java/JDI/jdi-mock-rest/src/main/java/com/epam/mock/web/ConfigurationHandlerController.java (hunk 11 above)
Comment (EvilXa0S) [reply Java/JDI/jdi-mock-rest/src/main/java/com/epam/mock/web/ConfigurationHandlerController.java:27]: Jn the other hand - this looks ok per story

Comment (EvilXa0S) [reply Java/JDI/jdi-mock-rest/src/main/java/com/epam/mock/web/ConfigurationHandlerController.java:27]: On the other hand this looks ok per story

**Review by EvilXa0S — APPROVED** [review]

File: Java/JDI/jdi-mock-rest/src/main/java/com/epam/mock/service/ConfigurationHandlerService.java [hunk 12]
```diff
@@ -0,0 +1,99 @@
...
+import java.util.Map;
+import org.springframework.stereotype.Service;
+
+/**
+ * Response service class.
+ */
+@Service
+public class ConfigurationHandlerService {
+
+    private static Map<String, ConfigurationHandler> configurationHandlerMap = new HashMap<>();
```
Comment (FunCat) [inline Java/JDI/jdi-mock-rest/src/main/java/com/epam/mock/service/ConfigurationHandlerService.java:15]: Replace String type to Enum Method?

**Pull request merged**